recipe/recipe.go

```diff
@@ -1,3 +1,66 @@
+/*
+Package 'recipe' implements actions mapping to YAML recipe.
+
+Recipe syntax
+
+Recipe is composed with 3 conditional parts:
+
+- definitions and expressions for text template engine
+(https://golang.org/pkg/text/template)
```

Comment:
This is weird. the templating isn't a part of the recipe it's a pre-processing step; The recipe is a Yaml file that gets processed through the templating engine.

Reply:
Does it matter for the end user?
Just imagine you know nothing about internals of debos and Go templating -- you'll see some strange things like '{{ var := value }}' in Yaml file -- which is a part of recipe because it is written in recipe file. May be it have sense to rephrase?

Reply:
I think templating is a quite well understood concept :) 

And yes it's mostly about rephrasing; I was quite confused when i read that paragraph, as i see the templating as just a method to generate the "actual" recipe file which has the requirements you describe further. 

  Having some basic examples of the go templating syntax is quite useful indeed. fwiw, we probalby want to document the `{{- $ospack := or .ospack "loqui-ospack.tar.gz" -}}` trick  for settings defaults as that's incredibly useful but also took some time to figure out :)

Reply:
Hm... I think rephrasing should be a part of #14 in that case